roborock/containers.py

```diff
@@ -306,144 +308,238 @@ class HomeDataDevice(RoborockBase):
+    is_new_ai_recognition_supported: bool = field(metadata={"new_feature_str_bit": NewFeatureStrBit.NEW_AI_RECOGNITION})
+    is_auto_collection_2_supported: bool = field(metadata={"new_feature_str_bit": NewFeatureStrBit.AUTO_COLLECTION_2})
+    is_right_brush_stretch_supported: bool = field(
+        metadata={"new_feature_str_bit": NewFeatureStrBit.RIGHT_BRUSH_STRETCH}
+    )
+    is_smart_clean_mode_set_supported: bool = field(
+        metadata={"new_feature_str_bit": NewFeatureStrBit.SMART_CLEAN_MODE_SET}
+    )
+    is_dirty_object_detect_supported: bool = field(
+        metadata={"new_feature_str_bit": NewFeatureStrBit.DIRTY_OBJECT_DETECT}
+    )
+    is_no_need_carpet_press_set_supported: bool = field(
+        metadata={"new_feature_str_bit": NewFeatureStrBit.NO_NEED_CARPET_PRESS_SET}
     )
+    is_voice_control_led_supported: bool = field(metadata={"new_feature_str_bit": NewFeatureStrBit.VOICE_CONTROL_LED})
+    is_water_leak_check_supported: bool = field(metadata={"new_feature_str_bit": NewFeatureStrBit.WATER_LEAK_CHECK})
+    is_min_battery_15_to_clean_task_supported: bool = field(
+        metadata={"new_feature_str_bit": NewFeatureStrBit.MIN_BATTERY_15_TO_CLEAN_TASK}
+    )
+    is_gap_deep_clean_supported: bool = field(metadata={"new_feature_str_bit": NewFeatureStrBit.GAP_DEEP_CLEAN})
+    is_object_detect_check_supported: bool = field(
+        metadata={"new_feature_str_bit": NewFeatureStrBit.OBJECT_DETECT_CHECK}
+    )
+    is_identify_room_supported: bool = field(metadata={"new_feature_str_bit": NewFeatureStrBit.IDENTIFY_ROOM})
+    is_matter_supported: bool = field(metadata={"new_feature_str_bit": NewFeatureStrBit.MATTER})
+
+    # is_multi_map_segment_timer_supported: bool = field(default=False)
+    # is_supported_drying_supported: bool = field(default=False)
+
+    @classmethod
+    def _is_new_feature_str_support(cls, o: int, new_feature_info_str: str) -> bool:
+        """
+        Checks feature 'o' in hex string 'new_feature_info_str'.
+        """
+        try:
+            l = o % 4
+            target_index = -((o // 4) + 1)
+            p = new_feature_info_str[target_index]
+            hex_char_value = int(p, 16)
+            is_set = (hex_char_value >> l) & 1
+            return bool(is_set)
+        except (IndexError, ValueError):
+            return False
+
+    @classmethod
+    def from_feature_flags(
+        cls, robot_new_features: int, new_feature_set: str, product_nickname: RoborockProductNickname
+    ) -> DeviceFeatures:
+        """Creates a DeviceFeatures instance from raw feature flags."""
+        unhexed_feature_info = int(new_feature_set[-8:], 16) if new_feature_set and len(new_feature_set) >= 8 else 0
+        upper_32_bits = robot_new_features // (2**32)
+
+        kwargs: dict[str, Any] = {}
+
+        for f in fields(cls):
+            if not f.metadata:
+                continue
+
+            if "robot_new_features" in f.metadata:
+                mask = f.metadata["robot_new_features"]
```

Comment:
Can we do something like this?

```suggestion
            if mask := f.metadata.get("robot_new_features"):
```